libetrv/data_struct.py

```diff
@@ -25,8 +26,73 @@ class ScheduleMode(enum.Enum):
     HOLD = 5
 
 
+class SettingsBitFieldData(BitFieldData):
+    LOCK_CONTROL = 7
+    VALVE_INSTALLED = 6
+    SLOW_REGULATION = 4
+    DISPLAY_FLIP = 3
+    VERTICAL_INSTALATION = 2
+    ADAPTABLE_REGULATION = 0
+
+    @property
+    def default_value(self):
+        return 0x22  # 2 Unknown bits maybe, but it failed if not set
+
+    @property
+    def lock_control(self):
+        return self._read_bit(SettingsBitFieldData.LOCK_CONTROL)
+
+    @lock_control.setter
+    def lock_control(self, value):
+        self._set_bit(SettingsBitFieldData.LOCK_CONTROL, value)
+
+    @property
+    def valve_installed(self):
+        return self._read_bit(SettingsBitFieldData.VALVE_INSTALLED)
+
+    @valve_installed.setter
+    def valve_installed(self, value):
+        self._set_bit(SettingsBitFieldData.VALVE_INSTALLED, value)
+
+    @property
+    def slow_regulation(self):
+        return self._read_bit(SettingsBitFieldData.SLOW_REGULATION)
+
+    @slow_regulation.setter
+    def slow_regulation(self, value):
+        self._set_bit(SettingsBitFieldData.SLOW_REGULATION, value)
+
+    @property
+    def display_flip(self):
+        return self._read_bit(SettingsBitFieldData.DISPLAY_FLIP)
+
+    @display_flip.setter
+    def display_flip(self, value):
+        self._set_bit(SettingsBitFieldData.DISPLAY_FLIP, value)
+
+    @property
+    def vertical_instalation(self):
+        return self._read_bit(SettingsBitFieldData.VERTICAL_INSTALATION)
+
+    @vertical_instalation.setter
+    def vertical_instalation(self, value):
+        self._set_bit(SettingsBitFieldData.VERTICAL_INSTALATION, value)
+
+    @property
+    def adaptable_regulation(self):
+        return self._read_bit(SettingsBitFieldData.ADAPTABLE_REGULATION)
+
+    @adaptable_regulation.setter
+    def adaptable_regulation(self, value):
+        self._set_bit(SettingsBitFieldData.ADAPTABLE_REGULATION, value)
+
+
 class SettingsData(eTRVData):
-    frost_protection_temperature = TemperatureField()
+    configuration_bits = BitField(
```

Comment:
We should avoid auto-save as much as possible because this process is very slow. A better idea is to gather a few options together and commit them later, for example with 5sec delay. Also when saving data to eTRV it will be stopped from normal work + connection to it can take like one minute. This Valve is not always available, so real-time monitoring and edition is not possible